### PR TITLE
 Add benchmark apps

### PR DESCRIPTION
**Pull request by JamesNK**
* BenchmarkClient - a test client that has some basic logic to measure RPS
* BenchmarkServer - will be used by ASP.NET perf environment (and by the BenchmarkClient on your own machine)
* NativeClient - has the same services as BenchmarkServer but powered by Grpc.Core

### CODE REVIEW

File: perf/Grpc.AspNetCore.Microbenchmarks/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+﻿using Grpc.AspNetCore.Microbenchmarks;
```
Comment (JunTaoLuo) [inline perf/Grpc.AspNetCore.Microbenchmarks/Properties/AssemblyInfo.cs]: nit: even AssemblyInfo needs copyright headers. Also to automatically add EOL, I've installed a VS/VS Code extension.

File: perf/benchmarkapps/BenchmarkClient/Program.cs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
```
Comment (JunTaoLuo) [inline perf/benchmarkapps/BenchmarkClient/Program.cs]: nit: sort

File: perf/benchmarkapps/BenchmarkServer/BenchmarkServer.csproj [hunk 3]
```diff
@@ -0,0 +1,42 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.0</TargetFramework>
+    <TargetFramework Condition="'$(BenchmarksTargetFramework)' != ''">$(BenchmarksTargetFramework)</TargetFramework>
+    <OutputType>Exe</OutputType>
+    <LangVersion>latest</LangVersion>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <IsTestAssetProject>true</IsTestAssetProject>
```
Comment (JunTaoLuo) [inline perf/benchmarkapps/BenchmarkServer/BenchmarkServer.csproj]: These are only needed in aspnet/AspNetCore, remove

File: perf/benchmarkapps/BenchmarkServer/hosting.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+  "urls": "",
+  "server.urls": "",
+  "UseTls": "false",
+  //"LogLevel": "Warning"
```
Comment (JunTaoLuo) [inline perf/benchmarkapps/BenchmarkServer/hosting.json]: ?

File: perf/benchmarkapps/BenchmarkServer/hosting.json [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{
+  "urls": "",
+  "server.urls": "",
+  "UseTls": "false",
```
Comment (JunTaoLuo) [inline perf/benchmarkapps/BenchmarkServer/hosting.json]: If we want to disable TLS, maybe we should just not specify a certificate instead?

File: perf/benchmarkapps/NativeServer/grpc-native.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
+{
+  "Default": {
+    "Client": "Grpc",
+    "ClientProperties": {
+      "UseTls": "false"
+    },
+
+    "Source": {
+      "Repository": "https://github.com/aspnet/grpc-dotnet.git",
```
Comment (JunTaoLuo) [inline perf/benchmarkapps/NativeServer/grpc-native.json]: This should be grpc/grpc-dotnet?

File: perf/benchmarkapps/BenchmarkServer/grpc-aspnetcore.json [hunk 7]
```diff
@@ -0,0 +1,27 @@
+{
+  "Default": {
+    "Client": "Grpc",
+    "ClientProperties": {
+      "UseTls": "false"
+    },
+
+    "Source": {
+      "Repository": "https://github.com/aspnet/grpc-dotnet.git",
```
Comment (JunTaoLuo) [inline perf/benchmarkapps/BenchmarkServer/grpc-aspnetcore.json]: grpc/grpc-dotnet? Also maybe the BranchOrCommit should also be changed to master upon merge?

File: perf/benchmarkapps/Shared/greet.proto [hunk 8]
```diff
@@ -0,0 +1,23 @@
+syntax = "proto3";
```
Comment (JunTaoLuo) [inline perf/benchmarkapps/Shared/greet.proto]: nit: copyright

File: perf/benchmarkapps/BenchmarkServer/hosting.json (hunk 5 above)
Comment (JamesNK) [reply perf/benchmarkapps/BenchmarkServer/hosting.json]: This is easily settable using the benchmark framework.

File: perf/Grpc.AspNetCore.Microbenchmarks/Properties/AssemblyInfo.cs [hunk 9]
```diff
@@ -16,6 +16,6 @@
 
 #endregion
 
-using Grpc.AspNetCore.Performance;
+using Grpc.AspNetCore.Microbenchmarks;
 
 [assembly: DefaultCoreConfigAttribute]
```
Comment (jtattermusch) [inline perf/Grpc.AspNetCore.Microbenchmarks/Properties/AssemblyInfo.cs]: nit: EOL

File: perf/benchmarkapps/BenchmarkClient/ConsoleOutLogger.cs [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+        public override ILogger ForType<T>()
+        {
+            if (typeof(T) == AssociatedType)
+            {
+                return this;
+            }
+            return new ConsoleOutLogger(typeof(T));
+        }
+    }
+}
```
Comment (jtattermusch) [inline perf/benchmarkapps/BenchmarkClient/ConsoleOutLogger.cs]: nit: EOL

**Pull request merged**